src/Assert.php

```diff
@@ -170,6 +170,49 @@ public static function integer(mixed $value, ?string $message = null): int
         return $value;
     }
 
+    /**
+     * Assert value is int and greater than limit
+     * @template       T
+     * @phpstan-assert numeric-string $value
+     * @phpstan-assert numeric-string $limit
+     *
+     * @param T                       $value
+     * @param T                       $limit
+     *
+     * @return T&numeric-string
+     */
+    public static function greaterThan(mixed $value, mixed $limit, ?string $message = null): mixed
```

Comment:
You're assuming here that the type of $value is the same as the type of $limit. This doesn't have to be the case when i call this with `Assert::greaterThan(1.0, 2)`. 

I don't think $limit needs to be a template variable. make this `int|float`.